includes/Core/Authentication/Authentication.php

```diff
@@ -362,10 +362,17 @@ function( $routes ) {
 			'googlesitekit_user_data',
 			function( $user ) {
 				if ( $this->profile->has() ) {
-					$profile_data              = $this->profile->get();
-					$user['user']['email']     = $profile_data['email'];
-					$user['user']['picture']   = $profile_data['photo'];
-					$user['user']['full_name'] = $profile_data['full_name'];
+					$profile_data            = $this->profile->get();
+					$user['user']['email']   = $profile_data['email'];
+					$user['user']['picture'] = $profile_data['photo'];
+					// Older versions of Site Kit (before n.e.x.t) did not
+					// fetch the user's full name, so we need to check for
+					// that attribute before using it.
+					//
+					// WP Cron will eventually update the profile data with
+					// this `full_name` attribute, but this prevents errors
+					// when a user upgrades to n.e.x.t but cron hasn't run yet.
```

Comment:
This is not _entirely_ accurate as we only use cron to fetch the profile data in the background if the request fails. See `OAuth_Client::refresh_profile_data`. We don't continuously poll for profile data in the background.

Aside from the cron fallback, we only fetch it when the user connects their Google account for the first time (see `OAuth_Client::authorize_user`).

We could consider refreshing the profile data more often – it could be a nice enhancement – I'll open an issue to do that.

Reply:
Oh, shoot, that makes sense, I forgot I added the functionality.

Yeah, in that case we should add that as a follow-up. I've adjusted the QA Brief again and will file a new issue.

Reply:
Filed #6003 as a follow-up.